core/api/src/services/mongoose/users.ts

```diff
@@ -40,6 +40,59 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const constructFilter = (userIds: UserId[], phoneCountryCodes: string[]) => {
+    const conditions: (
+      | { userId?: { $in: UserId[] } }
+      | { "phoneMetadata.countryCode": { $in: string[] } }
+    )[] = []
+
+    if (userIds.length > 0) {
+      conditions.push({ userId: { $in: userIds } })
+    }
+
+    if (phoneCountryCodes.length > 0) {
+      conditions.push({ "phoneMetadata.countryCode": { $in: phoneCountryCodes } })
```

Comment:
we have index on user id but this can be problematic in prod. I thinks is better to add an index on this

Reply:
added an index